modules/tinymce/src/core/main/ts/content/GetContentImpl.ts

```diff
@@ -22,9 +22,17 @@ const getContentFromBody = (editor: Editor, args: GetContentArgs, body: HTMLElem
   if (args.format === 'raw') {
     content = Tools.trim(TrimHtml.trimExternal(editor.serializer, body.innerHTML));
   } else if (args.format === 'text') {
-    // return empty string for text format when editor is empty to avoid bogus elements being returned in content
-    content = Zwsp.trim(body.innerText);
-    content = content === '\n' ? '' : content;
+    const doc = editor.getDoc();
+    content = body.innerHTML.replace(/<br data-mce-bogus="1">/g, Unicode.zeroWidth);
```

Comment:
This replacement string won't work if the content is XHTML, for example `<br data-mce-bogus="1"/>`.

Reply:
It also wouldn't catch cases where we have 2 attributes on a bogus element (granted I can't think of any off the top of my head). Either way, we probably want to consider doing something more like this: https://github.com/tinymce/tinymce/blob/236891ffcdf2fd1b698d020e67c31838fb110019/modules/tinymce/src/core/main/ts/dom/TrimHtml.ts#L90.